core/src/main/scala/kafka/network/RequestChannel.scala

```diff
@@ -53,6 +53,8 @@ object RequestChannel extends Logging {
     @volatile var apiLocalCompleteTimeMs = -1L
     @volatile var responseCompleteTimeMs = -1L
     @volatile var responseDequeueTimeMs = -1L
+    @volatile var apiRemoteCompleteTimeMs = -1L
```

Comment:
The volatile tag is needed because the readers are in the network thread and the writers are in the request handler threads or the purgatory threads. Not sure why apiLocalTimeCompleteTimeMs can be negative.